modules/imgcodecs/src/loadsave.cpp

```diff
@@ -60,18 +61,16 @@ namespace cv {
 
 // TODO Add runtime configuration
 #define CV_IO_MAX_IMAGE_PARAMS (50)
-#define CV_IO_MAX_IMAGE_WIDTH (1<<20)
-#define CV_IO_MAX_IMAGE_HEIGHT (1<<20)
-#define CV_IO_MAX_IMAGE_PIXELS (1<<30) // 1 Gigapixel
 
 static Size validateInputImageSize(const Size& size)
 {
     CV_Assert(size.width > 0);
-    CV_Assert(size.width <= CV_IO_MAX_IMAGE_WIDTH);
+    CV_Assert(static_cast<size_t>(size.width) <= utils::getConfigurationParameterSizeT("OPENCV_IO_MAX_IMAGE_WIDTH", 1<<20));
     CV_Assert(size.height > 0);
-    CV_Assert(size.height <= CV_IO_MAX_IMAGE_HEIGHT);
+    CV_Assert(static_cast<size_t>(size.height) <= utils::getConfigurationParameterSizeT("OPENCV_IO_MAX_IMAGE_HEIGHT", 1<<20));
     uint64 pixels = (uint64)size.width * (uint64)size.height;
-    CV_Assert(pixels <= CV_IO_MAX_IMAGE_PIXELS);
+    uint64 minSizeFile = utils::getConfigurationParameterSizeT("OPENCV_IO_MAX_IMAGE_WIDTH", 1 << 20)*utils::getConfigurationParameterSizeT("OPENCV_IO_MAX_IMAGE_HEIGHT", 1 << 20);
+    CV_Assert(pixels <= utils::getConfigurationParameterSizeT("OPENCV_IO_MAX_IMAGE_PIXELS", std::min<size_t>(minSizeFile, std::numeric_limits<std::size_t>::max()>>2)));
```

Comment:
1. `minSizeFile` is not needed. Max width, max height and max total number of pixels should stay different. Nobody want to allocate 128Gb on server due malformed input image.

2. `std::numeric_limits<std::size_t>::max()>>2` - this threshold is very huge, so please keep defaults the same.

Reply:
OK